projects/batfish/src/test/java/org/batfish/job/BatfishJobExecutorTest.java

```diff
@@ -0,0 +1,214 @@
+    Set<String> output = new HashSet<>();
+    List<BatfishException> failureCauses = new ArrayList<>();
+    BfTestAnswerElement ae = new BfTestAnswerElement();
+    executor.markJobCompleted();
+    executor.handleJobResult(bfTestResult, output, ae, failureCauses);
+
+    // checking the log of the executor for the job finished
+    assertEquals(
+        _logger.getHistory().toString(BatfishLogger.LEVEL_INFO),
+        executor.getSuccessMessage(bfTestResult));
+  }
+
+  @Test
+  public void testHandleJobResultFailure() {
+    Settings settings = new Settings();
+    settings.setHaltOnParseError(false);
+
+    // initializing executor
+    BatfishJobExecutor<BfTestJob, BfTestAnswerElement, BfTestResult, Set<String>> executor =
+        new BatfishJobExecutor<>(
+            settings, _logger, settings.getHaltOnParseError(), TEST_EXECUTOR_DESC);
+    executor.initializeJobsStats(Lists.newArrayList(new BfTestJob(settings, "result1")));
+
+    // Simulating failure of a job and handling the result
+    // initiating a separate logger from the Executor logger
+    BatfishLogger jobLogger = new BatfishLogger(BatfishLogger.LEVELSTR_INFO, false);
+    BfTestResult bfTestResult =
+        new BfTestResult(
+            TEST_JOB_DELAY,
+            jobLogger.getHistory(),
+            new BatfishException("Test Job Failure Message"));
+
+    Set<String> output = new HashSet<>();
+    List<BatfishException> failureCauses = new ArrayList<>();
+    BfTestAnswerElement ae = new BfTestAnswerElement();
+    executor.markJobCompleted();
+    executor.handleJobResult(bfTestResult, output, ae, failureCauses);
+
+    //checking that correct failure message is written in the log
+    assertEquals(failureCauses.get(0).getMessage(), executor.getFailureMessage(bfTestResult));
+  }
+
+  @Test
+  public void testHandleProcessingError() {
+    Settings settings = new Settings();
+    settings.setHaltOnParseError(true);
+
+    // initializing executor
+    BatfishJobExecutor<BfTestJob, BfTestAnswerElement, BfTestResult, Set<String>> executor =
+        new BatfishJobExecutor<>(
+            settings, _logger, settings.getHaltOnParseError(), TEST_EXECUTOR_DESC);
+    List<BatfishException> failureCauses = new ArrayList<>();
+
+    // checking if the exception thrown has correct class and message
+    _thrown.expect(HandleProcessingErrorException.class);
+    executor.handleProcessingError(
+        Lists.newArrayList(new BfTestJob(settings, "result1")), failureCauses);
+  }
+
+  private class BfTestJob extends BatfishJob<BfTestResult> {
+    private String _testValue;
+
+    public BfTestJob(Settings settings, String testValue) {
+      super(settings);
+      _testValue = testValue;
+    }
+
+    @Override
+    public BfTestResult call() throws Exception {
+      long startTime = System.currentTimeMillis();
+      long elapsedTime;
+      try {
+        TimeUnit.SECONDS.sleep(1);
```

Comment:
no real sleeps in tests. Can use a virtual clock if you are testing something that is accounting for the total time slept. (But I don't think that's how this was being used)

Reply:
removed